proposals/gc/MVP.md

```diff
@@ -266,10 +308,15 @@ TODO: Is 31 bit value range the right choice?
 
 #### Casts
 
-* `cast <reftype> <reftype>` casts a reference value down to a type given by a RTT representation
-  - `cast t t' : [t (rtt t')] -> [t']`
+* `ref.test <reftype> <reftype>` tests whether a reference value is of a type given by a RTT representation
```

Comment:
So here we have all 3 of: test-producing-i32, trapping-cast, branching-cast, but in the other cases (i32ref, optref, funcref) we only have different subsets of these.  Is that intentional or can we have the same 3 operations for all these basic reference types?

Reply:
For optref you actually have all three as well (ref.is_null preexists) -- optref is a bit special, since you want to be able to do these ops even when the type does not have RTT, so the RTT-based ops cannot be used.

For the other two, test and cast are merely shorthands for the generic ones that seem like they would come in handy. We could remove them. Inversely, we could also add a branching version for each, but that would be a much more heavyweight addition, so perhaps not justified as a mere shorthand.

But I don't have strong opinions either way.

Reply:
Not really, and it seems like we could switch later; it just seems preferable to have regularity between all the conceptual downcasts.

Reply:
Makes sense. I changed the text to list them as possibilities.